pyresample/gradient/__init__.py

```diff
@@ -406,3 +440,194 @@ def _concatenate_chunks(chunks):
     res = da.concatenate(res, axis=2)
 
     return res
+
+
+def _fill_in_coords(target_geo_def, data_coords, data_dims):
+    # TODO: this will crash when the target geo definition is a swath def.
+    x_coord, y_coord = target_geo_def.get_proj_vectors()
+    coords = []
+    for key in data_dims:
+        if key == 'x':
+            coords.append(x_coord)
+        elif key == 'y':
+            coords.append(y_coord)
+        else:
+            coords.append(data_coords[key])
+    return coords
+
+
+def ensure_data_array(func):
+    """Ensure the data is an instance of a an xarray.DataArray with correct dimensions."""
+    @wraps(func)
+    def wrapper(self, data, *args, **kwargs):
+        if not isinstance(data, xr.DataArray):
+            if data.ndim != 2:
+                raise TypeError("Use a xarray.DataArray to label the dimensions"
+                                " of arrays with other than two dimensions.")
+            else:
+                data = xr.DataArray(data, dims=["y", "x"])
+        dims = data.dims
+        data = data.transpose(..., "y", "x")
+        return func(self, data, *args, **kwargs).transpose(*dims)
+    return wrapper
+
+
+class RBGradientSearchResampler(BaseResampler):
+    """Resample using gradient search based bilinear interpolation."""
+
+    def __init__(self, source_geo_def, target_geo_def):
+        """Init GradientResampler."""
+        super().__init__(source_geo_def, target_geo_def)
+        logger.debug("/!\\ Instantiating an experimental GradientSearch resampler /!\\")
+        self.indices_xy = None
+
+    def precompute(self, **kwargs):
+        """Precompute resampling parameters."""
+        from pyresample.resampler import resample_blocks
```

Comment:
Can this be imported at the top of the module?

Reply:
yes, fixed